source/core/Queue.cpp

```diff
@@ -57,5 +57,15 @@ VIClump* VIClumpQueue::Dequeue()
 
     return head;
 }
-
+// size of queque
+int VIClumpQueue::size()
+{
+    int numberOfElements = 0;
+    VIClumpQueue* temp = this;
+    while (temp->_head != nullptr) {
+        temp->_head = temp->_head->_next;
```

Comment:
Wouldn't this change the _head in the VIClumpQueue you are trying to get a size of? temp here is just a pointer into the actual VIClumpQueue. Don't you need a copy of the Queue for this to work?